### DISCUSSION TITLE
Specify and Lock Poetry Version

### DISCUSSION THREAD

**Discussion by Davrosh** [python-poetry/poetry#7427] [Q&A]
Poetry makes it easy to pin specific versions of installed packages but is there a way to specify the version of Poetry itself my project is using either in the `pyproject.toml` file or somewhere else in the project?

Looking back at older examples, older versions of Poetry listed the version of Poetry used under `[build-system]` but now it only lists out `poetry-core`, the new build backend.
My Poetry 1.2.2 managed project has the following setup:
```
[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

If there is not such an option, what is the recommended way of enforcing a specific version of Poetry on a Poetry managed project between different members of the same team?

**Comment by dimbleby** [comment]
see discussion in #3316.  TLDR poetry currently provides no mechanism here.

**Comment by Davrosh** [comment]
Got it @dimbleby  - there is no current equivalent to `engines` or `engineStrict` from `package.json`. Thanks!

Still, as there are those who are transitioning from other language ecosystems that support this and would like to know how to maintain the version of Poetry used on a project between several team members, **perhaps you or some of the other maintainers could share how they are doing this on some of their projects**, whether it be by script or by the use of some third-party tool (like pyprojectx)?

I feel like this is something that has received notice (looking at the mentioned discussion #3316 ) but isn't being addressed enough right now on the docs.
